scripts/controller.gd

```diff
@@ -117,6 +118,22 @@ func _on_timer_timeout():
 	self._current_state = STATES.AWAITING_RESULT
 	self._board.turn_all_piece_lights_off()
 
+func _select_next_piece():
+	self._selected_piece_id = self._get_next_piece_id()
+	self._update_lights()
+
+func _get_next_piece_id() -> String:
+	var sorted = self._board.get_sorted_player_pieces()
+	if self._selected_piece_id == "":
+		return sorted[0]
+	var idx = sorted.find(self._selected_piece_id)
+	var next_idx = (idx + 1) % len(sorted)
+	return sorted[next_idx]
+
+func _update_lights():
+	self._board.turn_all_player_piece_lights_on(self._selectable_light_intensity)
+	self._board.get_piece_by_id(self._selected_piece_id).turn_light_on(self._selected_light_intensity)
```

Comment:
I guess this will also crash because `get_piece_by_id` returns `null` if there's no piece left.
```suggestion
	var piece := self._board.get_piece_by_id(self._selected_piece_id)
	if piece != null:
	    piece.turn_light_on(self._selected_light_intensity)
```